plugins/parsers/bindata/parser.go

```diff
@@ -0,0 +1,216 @@
+package bindata
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/metric"
+)
+
+const timeKey = "time"
+const timezone = "UTC"
+const defaultStringEncoding = "UTF-8"
+const defaultTimeFormat = "unix"
+
+// Field is a binary data field descriptor
+type Field struct {
+	Name string
+	Type string
+	Size uint
+}
+
+// BinData is a binary data parser
+type BinData struct {
+	metricName     string
+	timeFormat     string
+	endiannes      string
+	byteOrder      binary.ByteOrder
+	stringEncoding string
+	fields         []Field
```

Comment:
Please export these fields. Please also add toml-tags to those options to allow for the new parser format (see PR #8791 and the CSV parser as an example).